Doc/library/collections.abc.rst

```diff
@@ -272,6 +272,12 @@ Collections Abstract Base Classes -- Detailed Descriptions
       The index() method added support for *stop* and *start*
       arguments.
 
+   .. deprecated:: 3.12
+      The :class:`ByteString` ABC has been deprecated.
+      For use in typing, prefer a union, like ``bytes | bytearray``, or
+      :class:`collections.abc.Buffer`.
```

Comment:
Surprised CI doesn't complain that this class doesn't exist yet. I'll hopefully get it implemented soon.

Reply:
Here's the nit-picky warnings that https://github.com/python/cpython/pull/102513 will check for:

```console
$ make -C Doc html SPHINXERRORHANDLING=-n 2>&1 | grep collections.abc.rst | tee >(wc -l)
/Users/hugo/github/cpython/Doc/library/collections.abc.rst:76: WARNING: py:class reference target not found: D
/Users/hugo/github/cpython/Doc/library/collections.abc.rst:185: WARNING: py:meth reference target not found: object.__subclasshook__
/Users/hugo/github/cpython/Doc/library/collections.abc.rst:191: WARNING: py:meth reference target not found: __iter__
/Users/hugo/github/cpython/Doc/library/collections.abc.rst:191: WARNING: py:meth reference target not found: __getitem__
/Users/hugo/github/cpython/Doc/library/collections.abc.rst:204: WARNING: py:meth reference target not found: __contains__
/Users/hugo/github/cpython/Doc/library/collections.abc.rst:208: WARNING: py:meth reference target not found: __hash__
/Users/hugo/github/cpython/Doc/library/collections.abc.rst:212: WARNING: py:meth reference target not found: __len__
/Users/hugo/github/cpython/Doc/library/collections.abc.rst:216: WARNING: py:meth reference target not found: __call__
/Users/hugo/github/cpython/Doc/library/collections.abc.rst:220: WARNING: py:meth reference target not found: __iter__
/Users/hugo/github/cpython/Doc/library/collections.abc.rst:222: WARNING: py:meth reference target not found: __iter__
/Users/hugo/github/cpython/Doc/library/collections.abc.rst:222: WARNING: py:meth reference target not found: __getitem__
/Users/hugo/github/cpython/Doc/library/collections.abc.rst:242: WARNING: py:meth reference target not found: __reversed__
/Users/hugo/github/cpython/Doc/library/collections.abc.rst:262: WARNING: py:meth reference target not found: __iter__
/Users/hugo/github/cpython/Doc/library/collections.abc.rst:262: WARNING: py:meth reference target not found: __reversed__
/Users/hugo/github/cpython/Doc/library/collections.abc.rst:262: WARNING: py:meth reference target not found: index
/Users/hugo/github/cpython/Doc/library/collections.abc.rst:262: WARNING: py:meth reference target not found: __getitem__
/Users/hugo/github/cpython/Doc/library/collections.abc.rst:262: WARNING: py:meth reference target not found: __getitem__
/Users/hugo/github/cpython/Doc/library/collections.abc.rst:276: WARNING: py:class reference target not found: collections.abc.Buffer
/Users/hugo/github/cpython/Doc/library/collections.abc.rst:276: WARNING: py:class reference target not found: collections.abc.Buffer
/Users/hugo/github/cpython/Doc/library/collections.abc.rst:300: WARNING: py:meth reference target not found: __await__
/Users/hugo/github/cpython/Doc/library/collections.abc.rst:308: WARNING: py:meth reference target not found: __await__
/Users/hugo/github/cpython/Doc/library/collections.abc.rst:318: WARNING: py:meth reference target not found: __await__
/Users/hugo/github/cpython/Doc/library/collections.abc.rst:326: WARNING: py:meth reference target not found: __await__
/Users/hugo/github/cpython/Doc/library/collections.abc.rst:365: WARNING: py:meth reference target not found: __contains__
/Users/hugo/github/cpython/Doc/library/collections.abc.rst:365: WARNING: py:meth reference target not found: __iter__
/Users/hugo/github/cpython/Doc/library/collections.abc.rst:365: WARNING: py:meth reference target not found: __len__
/Users/hugo/github/cpython/Doc/library/collections.abc.rst:365: WARNING: py:meth reference target not found: __and__
/Users/hugo/github/cpython/Doc/library/collections.abc.rst:365: WARNING: py:meth reference target not found: isdisjoint
/Users/hugo/github/cpython/Doc/library/collections.abc.rst:397: WARNING: py:meth reference target not found: _from_iterable
/Users/hugo/github/cpython/Doc/library/collections.abc.rst:397: WARNING: py:meth reference target not found: _from_iterable
/Users/hugo/github/cpython/Doc/library/collections.abc.rst:408: WARNING: py:meth reference target not found: __le__
/Users/hugo/github/cpython/Doc/library/collections.abc.rst:408: WARNING: py:meth reference target not found: __ge__
/Users/hugo/github/cpython/Doc/library/collections.abc.rst:413: WARNING: py:meth reference target not found: _hash
/Users/hugo/github/cpython/Doc/library/collections.abc.rst:413: WARNING: py:meth reference target not found: __hash__
      34
```

Reply:
https://github.com/python/cpython/pull/102513 has now been merged and the CI now complains about the undocumented class.

Please could you add some docs for it?

Reply:
Jelle has a separate PR for that already :-)

Reply:
Ah yes, and I reviewed it! For reference: https://github.com/python/cpython/pull/102571